substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+
+			Ok(())
+		}
+
+		/// OPF vote removal logic
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be signed
+		///
+		/// ## Details
+		///
+		/// This extrinsic allows users to remove a casted vote, as long as it is within the
+		/// vote-casting period.
+		///
+		/// ### Parameters
+		/// - `project_id`: The account that will receive the reward.
+		///
+		/// ### Errors
+		/// - [`Error::<T>::NoProjectAvailable`]: No project found under this project_id
+		///  
+		/// ## Events
+		/// - [`Event::<T>::VoteRemoved { who, project_id }`]: User's vote successfully removed
+		#[pallet::call_index(4)]
+		#[pallet::weight(<T as Config>::WeightInfo::remove_vote(T::MaxProjects::get()))]
+		pub fn remove_vote(origin: OriginFor<T>, project_id: ProjectId<T>) -> DispatchResult {
+			let voter = ensure_signed(origin.clone())?;
+			// Get current voting round & check if we are in voting period or not
+			Self::period_check()?;
+
+			// Remove previous vote from Referendum
+			let infos = WhiteListedProjectAccounts::<T>::get(project_id.clone())
+				.ok_or(Error::<T>::NoProjectAvailable)?;
+			T::Conviction::try_remove_vote(&voter, infos.index.into())?;
+			// Removal action executed
+			Self::try_remove_vote(voter.clone(), project_id.clone())?;
+
+			Self::deposit_event(Event::<T>::VoteRemoved { who: voter, project_id });
+			Ok(())
+		}
+
+		/// OPF Reward Claim logic
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be signed
+		///
+		/// ## Details
+		///
+		/// From this extrinsic any user can claim a reward for a nominated/whitelisted project.
+		///
+		/// ### Parameters
+		/// - `project_id`: The account that will receive the reward.
+		///
+		/// ### Errors
+		/// - [`Error::<T>::InexistentSpend`]:Spend or Spend index does not exists
+		/// - [`Not Claiming Period`]: Still not in claiming period
+		///  
+		/// ## Events
+		/// Emits [`Event::<T>::RewardClaimed`] if successful for a positive approval.
+		#[pallet::call_index(5)]
+		#[pallet::weight(<T as Config>::WeightInfo::claim_reward_for(T::MaxProjects::get()))]
+		pub fn claim_reward_for(origin: OriginFor<T>, project_id: ProjectId<T>) -> DispatchResult {
+			let _caller = ensure_signed(origin)?;
+			let now = T::BlockNumberProvider::current_block_number();
+			let mut info = Spends::<T>::get(&project_id).ok_or(Error::<T>::InexistentSpend)?;
+			Self::pot_check(info.amount)?;
+			ensure!(!info.claimed, Error::<T>::AlreadyClaimed);
+
+			if now >= info.expire {
+				Spends::<T>::remove(&project_id);
+				Self::deposit_event(Event::ExpiredClaim {
+					expired_when: info.expire,
+					project_id: project_id.clone(),
+				});
+				Ok(())
+			} else {
+				// transfer the funds
+				info.claimed = true;
+				Spends::<T>::insert(project_id.clone(), info.clone());
+				Self::spend(info.amount, project_id.clone())?;
+				Self::deposit_event(Event::RewardClaimed {
+					amount: info.amount,
+					project_id: project_id.clone(),
+				});
+				WhiteListedProjectAccounts::<T>::remove(&project_id);
```

Comment:
As the spec, this shouldn't remove the project from the whitelisted projects. The project should be part of the next round too.